File: leetcode/442-find-all-duplicates-in-an-array.py
# https://leetcode.com/problems/find-all-duplicates-in-an-array/

from typing import List

class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        
        duplicates = []

        for i in range(len(nums)):
            abs_num = abs(nums[i])
            
            if nums[abs_num - 1] < 0:
                duplicates.append(abs_num)
            
            nums[abs_num - 1] *= -1
        
        return duplicates

File: leetcode/442-find-all-duplicates-in-an-array.js
// https://leetcode.com/problems/find-all-duplicates-in-an-array/

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
  const duplicates = [];

    for (let i = 0; i < nums.length; i += 1) {
      const absNum = Math.abs(nums[i]);
      if (nums[absNum-1] < 0) duplicates.push(absNum);
      nums[absNum-1] *= -1;
    }

  return duplicates;
};
